### PR TITLE
Send email from TWS Robots by default

### LINKED ISSUE #666: Change 'The Robots at Tea With Strangers'
although clever, when it comes to actually optimizing for what people see in their inboxes, 'Tea With Strangers Robots' would likely set context and increase open rates more effectively

screen shot 2015-12-07 at 3 40 29 pm

### PR DESCRIPTION
**Pull request by nickbarnwell**
Fixes #666 and #685

### CODE REVIEW

**Comment by nickbarnwell** [comment]
@ankitshah811 Also noticed all of the user registration emails come from Ankit, except for the Host Reg, which is from Robots. If anything, it should be the opposite, or – at the very least – consistent. Thoughts?

**Comment by ankitshah811** [comment]
@nickbarnwell let's make host reg consistent with the other ones. I'd like to keep the registration emails from a person and transactional emails (event confirmation, reminders, etc) from me

**Comment by nickbarnwell** [comment]
OK this is totally unclear to me and I want to close this PR out because it also has several URL fixes. These are all the emails we send currently:

* User Signs up for Tea With Strangers
  * User also registered for a tea time at the same time (Different Template)
* User Registers for a Tea Time (Confirmation)
* User Gets a _Default_ Tea Time Reminder (2 days and 12 hrs before; email is the same)
* User Gets a _Custom_ Tea Time Reminder
* User Gets Confirmation of Flaking
* User Gets Confirmation They Are On the Waitlist
* User Gets Notified that a Waitlisted Spot is Open
* User Gets Notified Tea Time was Cancelled
* User Gets A Follow Up Email After Tea Time for:
  * Flake
  * No Show
  * Waitlisted
  * Present

* Host Gets Reminded to Mark Tea Time Attendance
* Host Gets Confirmation of Tea Time Creation
* Hosts (plural) Get Confirmation Hosting Privileges Have Been Transferred
* Host Is Registered as a Host

Please let me know who FROM should be for each of these and I will fix as appropriate

**Comment by nickbarnwell** [comment]
I got about halfway through this commit and then realised we also need to update all of the mailer templates and copy to reflect that they're no longer coming from the TWS Robots. Will write up more tomorrow morning.

**Comment by ankitshah811** [comment]
* **Ankit** -- User Signs up for Tea With Strangers
* **? (Which email is this?)** -- User also registered for a tea time at the same time (Different Template)
* **TWS host via TWS** _(that is, robots@ sends, host email is reply to)_ -- User Registers for a Tea Time (Confirmation)
* **TWS host via TWS** -- User Gets a Default Tea Time Reminder (2 days and 12 hrs before; email is the same)
   * I don't think this email goes out anymore. I think we should reinstate it as default with option for custom tt reminder per host)
   * do we currently have code written to let hosts create tt-specific reminder emails? (That is, order of priority = email host custom wrote for specific tt > email host custom wrote for all their tts > default TWS reminder email) 
* TWS host via TWS -- User Gets a Custom Tea Time Reminder
* robots@ -- User Gets Confirmation of Flaking
* TWS host via TWS -- User Gets Confirmation They Are On the Waitlist
* TWS host via TWS -- User Gets Notified that a Waitlisted Spot is Open
* TWS host via TWS -- User Gets Notified Tea Time was Cancelled
   * ideally we can have hosts write a note here that turns into the cancellation email a la custom reminder emails. if not, I still think 'TWS host via TWS' is best
* User Gets A Follow Up Email After Tea Time for:
   * robots@ -- Flake
   * Ankit -- No Show
   * robots@ -- Waitlisted
   * robots@ -- Present
* robots@ -- Host Gets Reminded to Mark Tea Time Attendance
* robots@ -- Host Gets Confirmation of Tea Time Creation
* robots@ -- Hosts (plural) Get Confirmation Hosting Privileges Have Been Transferred
* robots@ -- Host Is Registered as a Host

Notes:

* robots@ from name should be 'TWS Robots'
* hosts should get an email every time someone signs up for their tt
* hosts should get an email every time someone cancels on their tt
* there's an email that is skipped over here — reminder to hosts 2d before tt to email attendees. this should be replaced with aforementioned reinstitution of attendee tt reminders

File: app/mailers/attendance_mailer.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def registration(attendance_id)
                                      content: IcalCreator.new(@tea_time).call.to_ical}
 
     mail(to: @attendance.user.email,
-         subject: "#{@attendance.user.name}! Pencil your tea time to your calendar!") do |format|
+         subject: "#{@attendance.user.name}! Pencil in your tea time on your calendar!") do |format|
```
Comment (ankitshah811) [inline app/mailers/attendance_mailer.rb]: Great catch 👍

File: app/views/attendance_mailer/flake.markerb [hunk 2]
```diff
@@ -11,5 +11,5 @@ It'd be awesome if you were able to find another time that does work for you. Be
```
Comment (ankitshah811) [inline app/views/attendance_mailer/flake.markerb]: <%= @tea_time.host.nickname %> here from Tea With Strangers. Just wanted to confirm that you canceled your seat at my tea time on <%= @tea_time.date_sans_year %>.

It's a shame that we won't get to share conversations, but you should try to get to another tea time. [See if something fits your schedule here](<%= url_for(controller: 'cities',
        action: 'show',
        id: @tea_time.city.city_code,
        only_path: false,
        utm_campaign: 'cancelspot',
        utm_medium: 'email',
        utm_content: 'show') %>).

File: app/views/attendance_mailer/flake.markerb [hunk 3]
```diff
@@ -11,5 +11,5 @@ It'd be awesome if you were able to find another time that does work for you. Be
 
 Hope your day is going beautifully. If not, grab a snack and watch this: <https://www.youtube.com/watch?v=l-gQLqv9f4o>.
 
-Bleep bleep bloop,  
-The Robots at Tea With Strangers
\ No newline at end of file
+Hope to see you soon!
```
Comment (ankitshah811) [inline app/views/attendance_mailer/flake.markerb]: ```Hope to see you at a tea time soon!```

**Comment by ankitshah811** [comment]
Have these comments been addressed in this PR? @nickbarnwell 

* TWS host via TWS -- User Gets a Default Tea Time Reminder (2 days and 12 hrs before; email is the same)
   * [TO DO] I don't think this email goes out anymore. I think we should reinstate it as default with option for custom tt reminder per host)
   * I think the email that is sent is a reminder to hosts 2d before tt to email attendees. 
      * [TO DO] this should be replaced with aforementioned reinstitution of these attendee tt reminders 
   * [TO DO] do we currently have code written to let hosts create custom tt-specific and host-specific reminder emails? (That is, order of priority = email host custom wrote for specific tt > email host custom wrote for all their tts > default TWS reminder email; refresher: http://decaffeinatedpizzafart.tumblr.com/post/126041965964/dispatch-no-1-july-29-2015)
      * @nickbarnwell If we have this, how much more complicated would it be to write the logic that allowed hosts to customize the emails below? If this is trivial to create + more of the work is in developing the UI for this, i'd like to make this possible and am happy to commit to designing/speccing the view from the past iteration this weekend to make this easy for hosts to take advantage of this
         * TWS host via TWS -- User Gets a Custom Tea Time Reminder
         * TWS host via TWS -- User Gets Confirmation of Flaking
         * TWS host via TWS -- User Gets Confirmation They Are On the Waitlist
         * TWS host via TWS -- User Gets Notified that a Waitlisted Spot is Open
         * TWS host via TWS -- User Gets Notified Tea Time was Cancelled

[TO DO] LMK if I should design this customized email UI 

* robots@ from name should be 'TWS Robots'
* [TO DO] hosts should get an email every time someone signs up for their tt
* [TO DO] hosts should get an email every time someone cancels on their tt

**Pull request merged**